controllers/sandbox_controller.go

```diff
@@ -141,5 +303,60 @@ func (r *SandboxReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&sandboxv1alpha1.Sandbox{}).
 		Owns(&corev1.Pod{}, builder.WithPredicates(labelSelectorPredicate)).
+		Owns(&corev1.Service{}, builder.WithPredicates(labelSelectorPredicate)).
 		Complete(r)
 }
+
+// ReconcileSandboxTTL will check if a sandbox has expired, and if so, delete it.
+// If the sandbox has not expired, it will requeue the request for the remaining time.
+
+func (r *SandboxReconciler) ReconcileSandboxTTL(ctx context.Context, sandbox *sandboxv1alpha1.Sandbox) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+	sandbox.Status.TTLExpiryTime = nil
+	if sandbox.Spec.TTL == nil || sandbox.Spec.TTL.TTLSeconds == 0 || sandbox.Spec.TTL.TTLFrom == "" {
+		log.Info("Sandbox TTL is not set, skipping TTL check.")
```

Comment:
Should we set TTLExpiryTime to nil in the status? Or because Sandbox CR is immutable, we just ignore the possibility that TTLExpiryTime needs to be set back to nil?

Reply:
good point .. may be we should reset it to nil

Reply:
done.